cosmic-applet-audio/src/main.rs

```diff
@@ -219,10 +219,8 @@ impl Application for Audio {
                         }
                     }
                 }
-                // TODO: view() should gray out buttons/slider when state is disconnected
                 pulse::Event::Disconnected => {
-                    println!("setting state to disconnected");
-                    self.pulse_state = PulseState::Disconnected
+                    panic!("Connection to pulseaudio failed. Exiting...");
```

Comment:
This is good enough for now, but once cosmic-panel adapts using `launch-pad` or does it's own applet-restarting, this will cause a non-zero exit code. So just logging the error and doing `std::process::exit(0)` might be cleaner?

Reply:
Oh, cosmic panel does use launchpad now

Reply:
Well in that case, this might cause a restart-loop?

Reply:
It might make sense to just try and reconnect each time the applet popup opens? That way if something unexpected happens users can just close and reopen the applet and maybe get functionality back, rather than requiring a reboot?
This should also keep the music control (via mpris I think?) working even if pulse doesn't

Reply:
Can media controls be useful without sound?

Would it be alright if launchpad were to enter a restart loop but have some kind of increasing delay between each restart?

Reply:
> Can media controls be useful without sound?

Sure, if the user is streaming spotify via their phone but controlling the stream via their computer

Reply:
> It might make sense to just try and reconnect each time the applet popup opens? That way if something unexpected happens users can just close and reopen the applet popup and maybe get functionality back, rather than requiring a reboot?

That sounds reasonable. So just hide or grey out the volume controls in that case, I suppose.

Reply:
Ok

Reply:
Sliders can't be disabled yet, so I just replace the sliders and buttons with "PulseAudio Disconnected" if it is in the disconnected state, but now the applet should gracefully handle disconnections & re-connections :)